.test-infra/jenkins/metrics_report/dashboards_parser.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+import re
+import unittest
+
+
+class Dashboard:
```

Comment:
All `.py` files in `sdks/python/apache_beam` directory are formatted with yapf. It would be good to do the same with this code.

Reply:
We can also consider adding your files to `triggerPathPatterns` inside https://github.com/apache/beam/blob/master/.test-infra/jenkins/job_PreCommit_PythonAutoformatter.groovy to make sure there won't be any style-breaking changes in the future.

Reply:
> All `.py` files in `sdks/python/apache_beam` directory are formatted with yapf. It would be good to do the same with this code.

I parse it with yapf as you suggested it.

> We can also consider adding your files to triggerPathPatterns inside https://github.com/apache/beam/blob/master/.test-infra/jenkins/job_PreCommit_PythonAutoformatter.groovy to make sure there won't be any style-breaking changes in the future.

I tested it and simple triggering triggerPathPatterns is not enough to check the code within .test-infra directory. 
It requires more changes which are not related to this PR so I created [BEAM-10823](https://issues.apache.org/jira/browse/BEAM-10823) with it.